src/Database/Redis/Cluster.hs

```diff
@@ -238,19 +284,34 @@ rawResponse (CompletedRequest _ _ r) = r
 -- cluster reconfiguration events, which should be rare.
 evaluatePipeline :: MVar ShardMap -> IO ShardMap -> Connection -> [[B.ByteString]] -> IO [Reply]
 evaluatePipeline shardMapVar refreshShardmapAction conn requests = do
-        shardMap <- hasLocked $ readMVar shardMapVar
-        requestsByNode <- getRequestsByNode shardMap
+        shardMap <- readMVar shardMapVar
```

Comment:
why `hasLocked` has been removed?
It would allow us to throw an exception if things are blocked indefinitely on `MVar`

Reply:
@neeraj97 added this back now